### PR TITLE
Correct permissions for `.ssh` and `authorized_keys`

### PR DESCRIPTION
**Pull request by wolfogre**
Set the correct permissions on the .ssh directory and authorized_keys file, or sshd will refuse to use them and lead to clone/push/pull failures.

It could happen when users have copied their data to a new volume and changed the file permission by accident, and it would be very hard to troubleshoot unless users know how to check the logs of sshd which is started by s6.

### CODE REVIEW

**Comment by wolfogre** [comment]
I got bitten by this problem, and it took me hours to figure it out ... 😭

**Review by lunny — APPROVED** [review]

**Review by denyskon — APPROVED** [review]

**Comment by wxiaoguang** [comment]
I think you also need to `chown`

**Comment by wolfogre** [comment]
> I think you also need to `chown`

https://github.com/go-gitea/gitea/blob/d17a848fe275c3e8734a4dfcaf2eae8ca0dc361c/docker/root/etc/s6/gitea/setup#L54-L57

**Comment by wxiaoguang** [comment]
IIRC, ssh only requires the private key to be "0600", for "authorized_keys" 0644 or 0640 is either fine.

And, there is a "/data/git/.ssh/environment" below, does it need proper mode too?

**Comment by wolfogre** [comment]
> IIRC, ssh only requires the private key to be "0600", for "authorized_keys" 0644 or 0640 is either fine.

I have tested it, you're right, but sshd will create `authorized_keys` with `0600`, I think it's a good idea to be consistent.

> Since the ".ssh" directory can have proper owner later, why "sshd will refuse"?

I don't get it, I think sshd will refuse if the file has a wrong permission even it has proper owner,  am I wrong?

> And, there is a "/data/git/.ssh/environment" below, does it need proper mode too?

I have tested it, `environment` doesn't matter, even `777`.

File: docker/root/etc/s6/gitea/setup [hunk 1]
```diff
@@ -2,7 +2,15 @@
...
-    chmod 700 /data/git/.ssh
+fi
+
+# Set the correct permissions on the .ssh directory and authorized_keys file,
+# or sshd will refuse to use them and lead to clone/push/pull failures.
+# It could happen when users have copied their data to a new volume and changed the file permission by accident,
+# and it would be very hard to troubleshoot unless users know how to check the logs of sshd which is started by s6.
+chmod 700 /data/git/.ssh
+if [ -f /data/git/.ssh/authorized_keys ]; then
+    chmod 600 /data/git/.ssh/authorized_keys
```
Comment (delvh) [inline docker/root/etc/s6/gitea/setup:13]: Do we also need to fix it in `rootless`?

**Review by delvh — APPROVED** [review]

File: docker/root/etc/s6/gitea/setup (hunk 1 above)
Comment (wolfogre) [reply docker/root/etc/s6/gitea/setup:13]: I couldn't find where.

Comment (wxiaoguang) [reply docker/root/etc/s6/gitea/setup:13]: rootless doesn't use OpenSSH so it doesn't need the authorized_keys

**Review by silverwind — APPROVED** [review]

**Comment by wolfogre** [comment]
> I was unable to create a backport for 1.20. @wolfogre, please send one manually. 🍵
> 
> ```
> go run ./contrib/backport 25721
> ...  // fix git conflicts if any
> go run ./contrib/backport --continue
> ```

Something was going wrong with @GiteaBot, it has already created a backport for 1.20 (#25730).

**Pull request merged**